store/postgres/src/store.rs

```diff
@@ -1346,7 +1333,28 @@ impl StoreTrait for Store {
         self: Arc<Self>,
         for_subscription: bool,
     ) -> Arc<(dyn QueryStore + Send + Sync + 'static)> {
-        Arc::new(crate::query_store::QueryStore::new(self, for_subscription))
+        use std::sync::atomic::Ordering;
+
+        let replica_id = match for_subscription {
```

Comment:
This is a bit nitpicky, but I would have written `if for_subscription { .. } else { .. }` here.

Reply:
The match does look nicer when the branches are single liners, but still I like being able to put the `false` case first when it is the common case, without using `!` to negate the condition.